data_diff/database.py

```diff
@@ -408,7 +436,11 @@ def connect_to_uri(db_uri: str, thread_count: Optional[int] = 1) -> Database:
             warehouse = dsn.query['warehouse']
         except KeyError:
             raise ValueError(f"Must provide warehouse. Expected format: '{HELP_SNOWFLAKE_URI_FORMAT}'")
-        return Snowflake(dsn.host, dsn.user, dsn.password, warehouse=warehouse, database=database, schema=schema)
+        try:
+            role = dsn.query['role']
+        except KeyError:
+            raise ValueError(f"Must provide role. Expected format: '{HELP_SNOWFLAKE_URI_FORMAT}'")
```

Comment:
Does the Snowflake constructor throw a similar error? I assume it is ok with out one for users that have a default role assigned.

That isn't true in our case and we'd like to pass that along, but maybe isn't quite the same as the other objects like warehouse, database, and scheme....... Not sure.

Reply:
This code is out-dated